### ISSUE TITLE
Improper Flake8 Behavior for TypeShed/Stub files

### ISSUE THREAD

**Issue by Holt59** [microsoft/vscode-flake8#161]
Type: <b>Bug</b>

# Behaviour
## Expected vs. Actual

Currently, Flake8 reports incorrect errors in VS Code when linting stub files (.pyi), e.g., regarding spaces between classes and functions, example in a `context.pyi` file:

```
# this produces 'multiple statements on one line (colon)'
class Context: ...
```

When running flake8 from the command line, the error does not occur because flake8 detects the `.pyi` file and switch "mode".

**Note:** I have tested with and without the flake8 extension and the behavior is the same which is why I'm opening the ticket here rather than on the flake8 extension.

## Steps to reproduce:

1. Create a file named `context.pyi` with the following content:

```
class Context: ...
```

# Diagnostic data

-   Python version (& distribution if applicable, e.g. Anaconda): 3.8.10
-   Type of virtual environment used (e.g. conda, venv, virtualenv, etc.): Venv
-   Value of the `python.languageServer` setting: Pylance

<details>

<summary>Output for <code>Python</code> in the <code>Output</code> panel (<code>View</code>→<code>Output</code>, change the drop-down the upper-right of the <code>Output</code> panel to <code>Python</code>)
</summary>

<p>

```
2023-06-08 11:07:03.772 [info] LINTING: Skipping linting from Python extension, since Flake8 extension is installed and enabled.
2023-06-08 11:07:03.819 [info] > ~\.envs\xxx\Scripts\python.exe ~\.vscode\extensions\ms-python.python-2023.8.0\pythonFiles\linter.py -m mypy --follow-imports=silent --ignore-missing-imports --show-column-numbers --no-pretty ~\Code\xxx\typings\context.pyi
2023-06-08 11:07:03.820 [info] cwd: ~\Code\xxx
2023-06-08 11:07:05.898 [info] ##########Linting Output - mypy##########

2023-06-08 11:07:05.898 [info] Success: no issues found in 1 source file

```

</p>
</details>

<details>

<summary>User Settings</summary>

<p>

```
Multiroot scenario, following user settings may not apply:

languageServer: "Pylance"

linting
• flake8Enabled: true
• mypyEnabled: true

formatting
• provider: "none"

```

</p>
</details>

Extension version: 2023.8.0
VS Code version: Code 1.78.2 (b3e4e68a0bc097f0ae7907b217c1119af9e03435, 2023-05-10T14:39:26.248Z)
OS version: Windows_NT x64 10.0.19045
Modes:
Sandboxed: Yes

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|11th Gen Intel(R) Core(TM) i7-11800H @ 2.30GHz (16 x 2304)|
|GPU Status|2d_canvas: enabled<br>canvas_oop_rasterization: disabled_off<br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>opengl: enabled_on<br>rasterization: enabled<br>raw_draw: disabled_off_ok<br>video_decode: enabled<br>video_encode: enabled<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled<br>webgpu: enabled|
|Load (avg)|undefined|
|Memory (System)|31.67GB (10.90GB free)|
|Process Argv|--crash-reporter-id 0a8d7cd3-aac1-42fb-9831-b63600030522|
|Screen Reader|no|
|VM|0%|
</details><details>
<summary>A/B Experiments</summary>

```
vsliv368:30146709
vsreu685:30147344
python383:30185418
vspor879:30202332
vspor708:30202333
vspor363:30204092
vslsvsres303:30308271
vserr242cf:30382550
pythontb:30283811
vsjup518:30340749
pythonptprofiler:30281270
vshan820:30294714
vstes263cf:30335440
vscod805:30301674
binariesv615:30325510
bridge0708:30335490
bridge0723:30353136
cmake_vspar411:30581797
vsaa593:30376534
pythonvs932:30410667
cppdebug:30492333
vsclangdf:30486550
c4g48928:30535728
dsvsc012cf:30540253
pynewext54:30695312
azure-dev_surveyone:30548225
vscccc:30610679
282f8724:30602487
pyind779:30671433
f6dab269:30613381
vscrp:30673768
pythonsymbol12:30671437
2i9eh265:30646982
showlangstatbar:30737416
vsctsb:30748421
pythonms35:30701012
03d35959:30757346
pythonfmttext:30731395
pythoncmv:30756943
fixshowwlkth:30730052
hidesbindicator:30760978
pythongtdpath:30739704
pythonnosm12tcf:30757130

```

</details>

**Comment by karthiknadig** [comment]
I get this same error running it from commandline:
![image](https://github.com/microsoft/vscode-flake8/assets/3840081/6271c9b9-4e53-44c9-8c11-138025807d13)

**Comment by Holt59** [comment]
> I get this same error running it from commandline: ![image](https://user-images.githubusercontent.com/3840081/245685916-6271c9b9-4e53-44c9-8c11-138025807d13.png)

Thanks, I don't know why I was not getting errors previously... I'll close this and re-open it if I can figure out how.
